scripts/finish-release.ts

```diff
@@ -0,0 +1,210 @@
+      if (branch) {
+        command.push(branch);
+      }
+    }
+    execSync(command.join(' '));
+  }
+
+  checkout(branch: string, create: boolean = false): void {
+    const command = ['git', 'checkout'];
+    if (create) {
+      command.push('-b');
+    }
+    command.push(branch);
+    execSync(command.join(' '));
+  }
+
+  merge(branch: string, options: Partial<MergeOptions> = {}): void {
+    const command = ['git', 'merge', branch];
+    if (options.message) {
+      command.push('-m');
+      command.push(`"${options.message}"`);
+    }
+    if (options.mode) {
+      command.push(`--${options.mode}`);
+    }
+    execSync(command.join(' '));
+  }
+
+  push(remote: string, branch: string) {
+    const command = ['git', 'push', remote, branch];
+    execSync(command.join(' '));
+  }
+
+  fetch(remote: string, branch?: string) {
+    const command = ['git', 'fetch', remote];
+    if (branch) {
+      command.push(branch);
+    }
+    execSync(`git fetch ${remote} ${branch}`);
+  }
+
+  isCleanWorkingTree(): boolean {
+    const buffer = execSync('git status --porcelain');
+    return !buffer.toString().trim();
+  }
+}
+
+export async function shouldContinue(read: rl.Interface, prompt: string): Promise<boolean> {
+  return new Promise<boolean>((resolve) => {
+    read.question(`${prompt}\nContinue? (y/n) `, (answer) => {
+      if (answer.toLowerCase().trim() === 'y') {
+        resolve(true);
+      } else {
+        resolve(false);
+      }
+    });
+  });
+}
+
+export function getCompareLink(devBranch: string, mergeBranch: string): string {
+  return `https://github.com/aws-amplify/amplify-cli/compare/${devBranch}...${mergeBranch}`;
+}
+
+export function finalize(upstreamName: string, devBranch: string, mergeBranch: string, releaseBranch: string, git: Git) {}
+
+export function prepareBranches(upstreamName: string, devBranch: string, mergeBranch: string, releaseBranch: string, git: Git) {
+  git.checkout(devBranch);
+  git.pull(upstreamName, devBranch);
+  git.checkout(mergeBranch, true);
+  git.fetch(upstreamName, releaseBranch);
+  git.merge(`${upstreamName}/${releaseBranch}`, { message: 'chore: merge release commit from main to dev' });
+}
+type Args = {
+  continue: boolean;
+};
+function getArgs(): Args {
+  const args = process.argv.slice(2);
+  return {
+    continue: args.length > 0 && args.indexOf('--continue') > -1,
+  };
+}
+
+process.on('uncaughtException', (err) => {
+  if (err.message === 'dirty working tree') {
+    process.exit(1);
+  }
+});
+
+export async function main() {
+  const args = getArgs();
+  const git = new Git();
+  const rlInterface = rl.createInterface({
+    input: stdin,
+    output: stdout,
+  });
+
+  if (!git.isCleanWorkingTree()) {
+    console.error('Please run the script from a clean working tree');
+    process.exit(2);
+  }
+
+  const releaseSha = git.getShortSha('main');
+  const DEV_BRANCH = 'dev';
+  const RELEASE_BRANCH = 'main';
+  const REPO_NAME = 'aws-amplify/amplify-cli';
```

Comment:
might be worth parametrizing all references to `aws-amplify/amplify-cli` in this script (with `aws-amplify/amplify-cli` as default). So that one can dry run scripts in our personal forks.
Btw. have you validated this script somehow?

Reply:
I added additional parameters. I verified behavior with this command:

`yarn finish-release --release-branch update-finish-release-script --repository sdstolworthy/amplify-cli`

Reply:
Could you share links to commit shas/test PR in the fork ?

Reply:
https://github.com/sdstolworthy/amplify-cli/pull/10